src/main/scala/beam/agentsim/agents/modalBehaviors/DrivingMode.scala

```diff
@@ -0,0 +1,132 @@
+package beam.agentsim.agents.modalBehaviors
+
+import akka.actor.ActorRef
+import beam.agentsim.agents.PersonAgent.{PersonData, ScheduleBeginLegTrigger}
+import beam.agentsim.agents.modalBehaviors.DrivingMode.CompleteLegTrigger
+import beam.agentsim.agents.modalBehaviors.DrivingMode._
+import beam.agentsim.agents.util.{AggregatorFactory, MultipleAggregationResult}
+import beam.agentsim.agents.vehicles.BeamVehicle._
+import beam.agentsim.agents.vehicles.VehicleData._
+import beam.agentsim.agents.vehicles.{EnterVehicleTrigger, LeaveVehicleTrigger, VehicleData}
+import beam.agentsim.agents.{BeamAgent, PersonAgent, TriggerShortcuts}
+import beam.agentsim.events.resources.vehicle._
+import beam.agentsim.scheduler.{Trigger, TriggerWithId}
+import beam.router.RoutingModel.BeamLeg
+import beam.sim.HasServices
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/**
+  * @author dserdiuk on 7/29/17.
+  */
+object DrivingMode {
+  case class CompleteLegTrigger(tick: Double, beamLeg: BeamLeg) extends Trigger
+  implicit val beamLegOrdering: Ordering[BeamLeg] = Ordering.by(_.startTime)
+
+}
+
+trait DrivingMode extends BeamAgent[PersonData] with TriggerShortcuts with HasServices with AggregatorFactory {
+  this: PersonAgent =>
+
+  type PassengerLogStorage = mutable.TreeMap[BeamLeg, mutable.ListBuffer[ReservationLogEntry]]
+
+  case class ReservationLogEntry(departFrom: BeamLeg, arriveAt: BeamLeg,  passenger: ActorRef) {
```

Comment:
I realize that functionally they can be used to represent the same thing, but BeamLegs might contain relatively large amounts of data (because they can have a GraphPath in them) which will make equality comparisons MUCH more expensive than a simple Id which is all we really need.